tests/apollo/util/bft.py

```diff
@@ -820,22 +844,20 @@ def start_replica(self, replica_id):
             raise AlreadyRunningError(replica_id)
 
         is_external = self.is_existing and self.config.stop_replica_cmd is not None
-        start_cmd = self.start_replica_cmd(replica_id)
-        replica_binary_path = start_cmd[0]
+        start_cmd, replica_binary_path = self.start_replica_cmd(replica_id)
         digest = self.binary_digest(replica_binary_path) if Path(replica_binary_path).exists() else 'Unknown'
-
         with log.start_action(action_type="start_replica_process", replica=replica_id, is_external=is_external,
                               binary_path=replica_binary_path, binary_digest=digest, cmd=' '.join(start_cmd)):
             my_env = os.environ.copy()
             my_env["RID"] = str(replica_id)
             if is_external:
-                self.procs[replica_id] = subprocess.run(
+                proc = subprocess.run(
```

Comment:
I believe that the code which you put in verify_matching_replica_client_communication should be in here.
Meaning that this function should return with you having an inner dictionary with all of the correct child processes.
The verify_matching_replica_client_communication function's name does not describe the modifications to the dictionary.

Reply:
We should probably add a new task for that.
I didn't touch the verify_matching_replica_client_communication function logic or tried to refactor it. At most, I did some name changes.